src/Tvheadend.cpp

```diff
@@ -1917,9 +1907,64 @@ void CTvheadend::ParseRecordingAddOrUpdate ( htsmsg_t *msg, bool bAdd )
   rec.SetStop(stop);
 
   /* Channel is optional, it may not exist anymore */
-  if (!htsmsg_get_u32(msg, "channel", &channel))
+  if (!htsmsg_get_u32(msg, "channel", &channel) && channel > 0)
+  {
+    /* Channel Id */
     rec.SetChannel(channel);
 
+    auto cit = m_channels.find(rec.GetChannel());
+    if (cit != m_channels.cend())
+    {
+      /* Channel type */
+      rec.SetChannelType(cit->second.GetType());
+
+      /* Channel name */
+      rec.SetChannelName(cit->second.GetName());
+    }
+  }
+
+  /* Channel type fallback (in case channel was deleted) */
+  if (!rec.GetChannelType() && m_conn.GetProtocol() >= 25)
+  {
+    htsmsg_t *files, *streams;
+    if ((files = htsmsg_get_list(msg, "files")) != NULL)
+    {
+      htsmsg_field_t *file, *stream;
+      uint32_t hasAudio = 0, hasVideo  = 0, u32;
```

Comment:
Please remove one of the two spaces between <code>hasVideo</code> and <code>= 0</code>.